File: bench/03-stats.ts
function total(ns: number[]): number {
  return ns.reduce((acc, x) => acc + x, 0);
}

function mean(ns: number[]): number {
  return total(ns) / ns.length;
}

function maximum(ns: number[]): number {
  return ns.reduce((acc, x) => x > acc ? x : acc, 0);
}

function minimum(ns: number[]): number {
  return ns.reduce((acc, x) => x < acc ? x : acc, 999999);
}

function countIf(ns: number[], pred: (x: number) => boolean): number {
  return ns.filter(pred).length;
}

File: bench/03-stats.py
from functools import reduce

def total(ns: list[int]) -> int:
    return reduce(lambda acc, x: acc + x, ns, 0)

def mean(ns: list[int]) -> int:
    return total(ns) // len(ns)

def maximum(ns: list[int]) -> int:
    return reduce(lambda acc, x: x if x > acc else acc, ns, 0)

def minimum(ns: list[int]) -> int:
    return reduce(lambda acc, x: x if x < acc else acc, ns, 999999)

def count_if(ns: list[int], pred) -> int:
    return len([x for x in ns if pred(x)])
